alacritty/src/display/mod.rs

```diff
@@ -1061,10 +1061,15 @@ impl Display {
     #[inline]
     fn damage_highlighted_hints<T: EventListener>(&self, terminal: &mut Term<T>) {
         let display_offset = terminal.grid().display_offset();
+        let last_line = terminal.screen_lines() - 1;
         for hint in self.highlighted_hint.iter().chain(&self.vi_highlighted_hint) {
-            for point in (hint.bounds.start().line.0..=hint.bounds.end().line.0).flat_map(|line| {
-                term::point_to_viewport(display_offset, Point::new(Line(line), Column(0)))
-            }) {
+            for mut point in
+                (hint.bounds.start().line.0..=hint.bounds.end().line.0).flat_map(|line| {
+                    term::point_to_viewport(display_offset, Point::new(Line(line), Column(0)))
+                })
+            {
+                // Clamp hint point to viewport bounds, since it could be off-screen.
+                point.line = point.line.clamp(0, last_line);
```

Comment:
Feels like we should just skip this point, rather than clamping?

Reply:
Hm, makes sense.